### PR TITLE
Extract PTY pending output store

### PR DESCRIPTION
**Pull request by sam-phinizy**
## Summary
- add a generic Tauri-free pending PTY output attachment store in roux-runtime
- move pending output insert/remove/retain bookkeeping out of PtyManager
- preserve Tauri ownership of channels and output delivery

## Verification
- cargo test -p roux-runtime pty_pending_output
- CI=1 cargo test -p roux pty::
- cargo test -p roux-runtime
- CI=1 cargo clippy -p roux --all-targets -- -D warnings

<h3>Greptile Summary</h3>

This PR has two independent workstreams: it extracts `PtyPendingOutput<Channel>` as a Tauri-free generic store in `roux-runtime`, and it refactors the xterm input-gating strategy from toggling `disableStdin` to using a permanently-attached custom key handler with an `inputEnabled` flag — preserving terminal protocol responses on unfocused panes while still blocking keyboard input.

- **Rust side**: `PtyPendingOutput<Channel>` wraps a `HashMap` with typed `insert`/`remove`/`retain_existing` and diagnostic helpers (`len`, `is_empty`, `contains_key`). `PtyManager` is updated mechanically; all PTY bookkeeping behavior is preserved.
- **Frontend — xterm input model**: `disableStdin` is now always `false`; a single `attachCustomKeyEventHandler` installed at construction time gates keyboard events through `this.inputEnabled`. `setInputEnabled` sets the field instead of mutating `terminal.options.disableStdin`.
- **Frontend — focus capture**: `onmousedowncapture` on the pane shell claims logical focus in the capture phase so stdin routing follows the pane even when xterm stops event propagation.

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge; the Rust extraction is a mechanical refactor and the xterm input-gating change is validated by new tests.

Both workstreams are well-scoped: the Rust side is a pure rename/delegation with no behavior change, and the frontend input-gating refactor is directly exercised by the new xtermController test that explicitly checks disableStdin stays false and keyboard events are blocked via the handler. The capture-phase focus handler addresses a real propagation edge case. Only minor style cleanups remain.

PaneShell.svelte has a redundant double setLogicalFocus in handleDrop and a stale disableStdin comment, but neither affects correctness.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| crates/roux-runtime/src/pty_pending_output.rs | New generic PtyPendingOutput<Channel> store wrapping HashMap; clean API with insert/remove/retain_existing + diagnostic helpers and comprehensive unit tests. |
| crates/roux-runtime/src/lib.rs | Adds pty_pending_output module registration; trivial one-line addition, no issues. |
| src-tauri/src/pty.rs | Swaps HashMap for PtyPendingOutput in PtyManager; retain() call becomes retain_existing(); clean mechanical refactor with no behavior change. |
| src/lib/panes/xtermController.ts | Replaces disableStdin toggling with an inputEnabled flag in a single always-attached custom key handler; disableStdin stays false so terminal protocol responses (OSC etc.) are never suppressed. |
| src/lib/panes/focus.ts | requestDomFocus now delegates to setLogicalFocus instead of only calling terminal.focus(); callers in handleDrop that explicitly call both functions now have a redundant first call. |
| src/lib/components/PaneShell.svelte | Adds onmousedowncapture to claim logical focus before xterm stops event propagation; handleDrop now redundantly calls setLogicalFocus twice (explicit call + through requestDomFocus); stale comment references disableStdin. |
| src/lib/panes/__tests__/focus.test.ts | Adds requestDomFocus test covering new logical-focus side effect; existing test still passes as a subset of new behavior. |
| src/lib/panes/__tests__/xtermController.test.ts | Improves Terminal mock to capture constructor options; adds test verifying setInputEnabled gates keyboard events without touching disableStdin. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[User mousedown on PaneShell] --> B{capture phase onmousedowncapture}
    B --> C[setLogicalFocus paneId]
    C --> D[focusedPaneId store updated]
    C --> E[setInputEnabled true on target / false on others]
    C --> F[terminal.focus on target]
    B --> G{xterm stops propagation?}
    G -- Yes --> H[handleMouseDown skipped]
    G -- No --> I[handleMouseDown bubble phase]
    I --> C
    subgraph XtermController
        J[customKeyEventHandler] --> K{inputEnabled?}
        K -- false --> L[return false: block event]
        K -- true --> M{customKeyHandler set?}
        M -- No --> N[return true: allow event]
        M -- Yes --> O[delegate to customKeyHandler]
    end
    E --> J
    subgraph PtyManager
        P[pending_outputs: PtyPendingOutput] --> Q[insert / remove / retain_existing]
    end
```

<a href="https://app.greptile.com/ide/claude-code?prompt=Fix%20the%20following%202%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%202%0Asrc%2Flib%2Fcomponents%2FPaneShell.svelte%3A327-328%0ARedundant%20%60setLogicalFocus%60%20call%3A%20%60requestDomFocus%60%20now%20delegates%20to%20%60setLogicalFocus%60%20internally%2C%20so%20the%20explicit%20call%20on%20the%20line%20above%20is%20a%20no-op%20duplicate.%20The%20two-step%20pattern%20made%20sense%20before%20%28logical%20focus%2C%20then%20DOM%20focus%20were%20distinct%20operations%29%2C%20but%20they're%20now%20equivalent.%20The%20first%20call%20can%20be%20removed.%0A%0A%60%60%60suggestion%0A%20%20%20%20requestDomFocus%28paneId%29%3B%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%202%0Asrc%2Flib%2Fcomponents%2FPaneShell.svelte%3A469-470%0AStale%20comment%3A%20%60setInputEnabled%60%20no%20longer%20toggles%20%60disableStdin%60%20%E2%80%94%20it%20now%20sets%20the%20%60inputEnabled%60%20field%20that%20gates%20the%20custom%20key%20handler.%20The%20%60disableStdin%60%20option%20stays%20%60false%60%20permanently%20after%20this%20refactor%2C%20so%20the%20parenthetical%20is%20misleading%20for%20the%20next%20reader.%0A%0A%60%60%60suggestion%0A%20%20%2F%2F%20Focus%20effect%3A%20refit%20when%20gaining%20logical%20focus%20%28inputEnabled%20just%20changed%2C%0A%20%20%2F%2F%20terminal%20may%20need%20resize%29.%0A%60%60%60%0A%0A&repo=phin-tech%2Froux&pr=191&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=3">Fix All in Claude Code</picture></a> <a href="https://chatgpt.com/codex/deeplink?prompt=IMPORTANT%3A%20Work%20in%20the%20repository%20%22phin-tech%2Froux%22%20on%20the%20existing%20branch%20%22feature%2Fextract-pty-pending-output%22.%20Checkout%20that%20branch%20%E2%80%94%20do%20NOT%20create%20a%20new%20branch%20or%20open%20a%20new%20PR.%20Push%20your%20changes%20to%20%22feature%2Fextract-pty-pending-output%22.%0A%0AFix%20the%20following%202%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%202%0Asrc%2Flib%2Fcomponents%2FPaneShell.svelte%3A327-328%0ARedundant%20%60setLogicalFocus%60%20call%3A%20%60requestDomFocus%60%20now%20delegates%20to%20%60setLogicalFocus%60%20internally%2C%20so%20the%20explicit%20call%20on%20the%20line%20above%20is%20a%20no-op%20duplicate.%20The%20two-step%20pattern%20made%20sense%20before%20%28logical%20focus%2C%20then%20DOM%20focus%20were%20distinct%20operations%29%2C%20but%20they're%20now%20equivalent.%20The%20first%20call%20can%20be%20removed.%0A%0A%60%60%60suggestion%0A%20%20%20%20requestDomFocus%28paneId%29%3B%0A%60%60%60%0A%0A%23%23%23%20Issue%202%20of%202%0Asrc%2Flib%2Fcomponents%2FPaneShell.svelte%3A469-470%0AStale%20comment%3A%20%60setInputEnabled%60%20no%20longer%20toggles%20%60disableStdin%60%20%E2%80%94%20it%20now%20sets%20the%20%60inputEnabled%60%20field%20that%20gates%20the%20custom%20key%20handler.%20The%20%60disableStdin%60%20option%20stays%20%60false%60%20permanently%20after%20this%20refactor%2C%20so%20the%20parenthetical%20is%20misleading%20for%20the%20next%20reader.%0A%0A%60%60%60suggestion%0A%20%20%2F%2F%20Focus%20effect%3A%20refit%20when%20gaining%20logical%20focus%20%28inputEnabled%20just%20changed%2C%0A%20%20%2F%2F%20terminal%20may%20need%20resize%29.%0A%60%60%60%0A%0A"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodexDark.svg?v=3"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInCodex.svg?v=3">Fix All in Codex</picture></a>

<sub>Reviews (6): Last reviewed commit: ["Keep terminal protocol replies enabled o..."](https://github.com/phin-tech/roux/commit/b23c88d8fd23879c848147af7ee78657e0eb686c) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=33283914)</sub>

### CODE REVIEW

File: crates/roux-runtime/src/pty_pending_output.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    #[cfg(test)]
+    fn len(&self) -> usize {
+        self.pending.len()
+    }
+
+    #[cfg(test)]
+    fn contains_key(&self, pty_id: &str) -> bool {
+        self.pending.contains_key(pty_id)
+    }
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply crates/roux-runtime/src/pty_pending_output.rs]: Fixed in edaf7bc. Exposed len, is_empty, and contains_key as public diagnostics helpers on PtyPendingOutput.

**Comment by sam-phinizy** [comment]
Superseded by #193, which now targets main and contains the full stack.
